servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/OpenTelemetryHttpRequestFilter.java

```diff
@@ -111,12 +121,19 @@ private Single<StreamingHttpResponse> trackRequest(final StreamingHttpRequester
         Single<StreamingHttpResponse> response;
         try {
             propagators.getTextMapPropagator().inject(Context.current(), request.headers(),
-                    HeadersPropagatorSetter.INSTANCE);
+                HeadersPropagatorSetter.INSTANCE);
             response = delegate.request(request);
         } catch (Throwable t) {
             tracker.onError(t);
             return Single.failed(t);
         }
         return tracker.track(response);
     }
+
+    private String getSpanName(StreamingHttpRequest request) {
+        if (componentName != null && !componentName.trim().isEmpty()) {
+            return componentName;
+        }
+        return Optional.ofNullable(request.host()).orElse(request.requestTarget());
```

Comment:
Consider just using regular if / else with null, I don't think we need to construct an Optional instance here since it gets discarded anyways.

Reply:
switched to an if